### DISCUSSION TITLE
Help: HALFpipe appends /ext/ to beginning of working directory

### DISCUSSION THREAD

**Discussion by dastout** [HALFpipe/HALFpipe#351] [General] [answered]
Hello,

We are trying to learn HALFpipe and are encountering a problem. We specify the working directory but when it adds /ext/ to the beginning of our file path. We do not have an /ext/ directory, so we are not sure how this is happening. Note that we are trying to run HALFpipe from a mount. Could this be causing the problem?

Thanks,

Danny

**Comment by HippocampusGirl** [comment, marked as answer]
Hi Danny,

Since `HALFpipe` is running as a container, it cannot access your computer's files by default. That is why in the instructions, we ask users to give it access to their files via the `/ext` path, for example via the command 

```bash
singularity run --contain --cleanenv --bind /:/ext halfpipe-halfpipe-latest.simg
```

If you want to manually pass a specific data directory to the container, you can disable this behavior by adding the `--fs-root /` command line flag when you start `HALFpipe`.

**Comment by dastout** [comment]
Hi Lea,

Thank you for your response. I'm running it via Docker on a MacBook. I started HALFpipe using this command:
docker run --interactive --tty --volume /:/ext halfpipe/halfpipe

Is there anything different that I need to do instead?

Thanks,

Danny

--
Daniel M. Stout, Ph.D.
Research Health Scientist | VA San Diego Healthcare System
Assistant Professor | Department of Psychiatry UC San Diego
email: ***@***.***
________________________________

**Comment by HippocampusGirl** [comment]
Hi Danny,

That looks fine! What kind of issues are you experiencing?

**Comment by dastout** [comment]
I should note that I am trying this on two different computers. One is a Linux running Ubuntu on a mounted server. Then I'm also trying to run it on a local Mac OS. Some of my problems were on the Linux box. I plan to try again on Friday.

However, for the Mac, I'm still having problems with entering event timing files. It seems to acknowledge finding files but won't progress to next step. I'm not sure if it is because I selected BIDS format or how I am running HALPpipe via Docker:

[cid:11ff1d28-556b-4df1-b083-fd0f71a3c111]


Any advice or suggestions would be welcomed!

Thanks,

Danny

________________________________

**Reply by HippocampusGirl** [reply]
Hi  @dastout,

I think I was able to fix this bug yesterday. Could you try [with the newest version](https://github.com/HALFpipe/HALFpipe#download)?

**Comment by dastout** [comment]
Hi Leah,

After I pull in the latest version, I am getting an error when trying to load HALFpipe:

[cid:77ed3498-1d86-4cce-bc7b-c8a6a1a51357]

Any suggestions?

Thanks,

Danny
________________________________

**Comment by dastout** [comment]
Hi Leah,

An update to my last error. I ran the --debug command to provide a better output of the issue that I am now having with trying to load HALFpipe after installing the latest version:
[cid:89edd391-fff4-42e6-9dde-b6a1d8520ac4]

Thanks,

Danny
________________________________

**Reply by HippocampusGirl** [reply]
Hi @dastout,

Could you upload (or e-mail me to <enigma@charite.de>) your log files?

**Comment by dastout** [comment]
Hi Leah,

After I pulled the latest version, I am no longer able to even run halfpipe and I get that error message when I execute the run halfpipe on the command line. So, there are no log files created.

Thanks,

Danny
________________________________

**Reply by HippocampusGirl** [reply]
Oh wow, there seems to have been an issue with the container build and installing all the required packages. It's fixed now!

**Comment by dastout** [comment]
Hi Lea,

Thanks for fixing the bug! It seems to be running now. I had to use the non-BIDS option at the beginning of the halfpipe setup to load the data files and for it to accept the event timing text files.

I am using Docker for running halfpipe. I see that Docker has an option allowing me to increase the CPUs and memory. Should I increase these prior to running halfpipe?

Thanks,

Danny
________________________________

**Reply by HippocampusGirl** [reply]
Yes, I think you should have at least 8GB of memory available, and more CPUs may make it faster to run.

**Comment by dastout** [comment]
Sounds good. I boosted the CPU and memory via Docker. I'm not sure if it is helping much. I'm wondering if the memory setting in Docker is memory per CPU? Anyways, it seems to be running. Although this step is quite slow, it has been on this step for a couple of subjects for several hours:

[cid:c32a41ad-cb16-49e7-9f64-b3e6c310339c]


Thanks,

Danny
________________________________

**Reply by HippocampusGirl** [reply]
Hi @dastout, I don't think I can see your screenshot. Could you e-mail me at <enigma@charite.de>?

![image](https://user-images.githubusercontent.com/789054/177980490-55d7abba-b816-44a2-952f-5aac08096231.png)
